py/conftest.py

```diff
@@ -108,27 +109,214 @@ def get_driver_class(driver_option):
+            return self._request.config.option.binary
+        return None
+
+    @property
+    def browser_args(self):
+        if self._request.config.option.args:
+            return self._request.config.option.args
+        return None
+
+    @property
+    def driver_path(self):
+        if self._request.config.option.executable:
+            return self._request.config.option.executable
+        return None
+
+    @property
+    def headless(self):
+        self._headless = bool(self._request.config.option.headless)
+        if self._headless:
+            return True
+        return False
+
+    @headless.setter
+    def headless(self, driver_class):
+        if self.headless:
+            if driver_class.lower() == "chrome" or driver_class.lower() == "edge":
+                self._options.add_argument("--headless=new")
+            if driver_class.lower() == "firefox":
+                self._options.add_argument("-headless")
+
+    @property
+    def bidi(self):
+        return self._bidi
+
+    @bidi.setter
+    def bidi(self, value):
+        self._bidi = value
+        if self._bidi:
+            self._options.web_socket_url = True
+            self._options.unhandled_prompt_behavior = "ignore"
+
+    @property
+    def options(self):
+        return self._options
+
+    @options.setter
+    def options(self, cls_name):
+        if cls_name.lower() not in self.supported_options:
+            raise AttributeError(f"Invalid Options class {cls_name.lower()}")
+        self._options = getattr(webdriver, getattr(self.supported_options, cls_name.lower()))()
+        if self.driver_class == self.supported_drivers.firefox:
+            # There are issues with window size/position when running Firefox
+            # under Wayland, so we use XWayland instead.
+            os.environ["MOZ_ENABLE_WAYLAND"] = "0"
+        if self.driver_class == self.supported_drivers.remote:
+            self._options.set_capability("moz:firefoxOptions", {})
+            self._options.enable_downloads = True
+            os.environ["MOZ_ENABLE_WAYLAND"] = "0"
+        if self.driver_class == self.supported_drivers.webkitgtk:
+            self._options.overlay_scrollbars_enabled = False
+
+    @property
+    def service(self):
+        executable = self.driver_path
+        if executable:
+            module = getattr(webdriver, self.driver_class.lower())
+            self._service = module.service.Service(executable_path=executable)
+            return self._service
+        return None
+
+    @property
+    def driver(self):
+        self._driver = self._initialize_driver()
+        return self._driver
+
+    @property
+    def is_platform_valid(self):
+        if self.driver_class.lower() == "safari" and self.exe_platform != "Darwin":
+            return False
+        if self.driver_class.lower() == "ie" and self.exe_platform != "Windows":
+            return False
+        if "webkit" in self.driver_class.lower() and self.exe_platform != "Linux":
```

Comment:
WebKitGTK and WPEWebKit can run on other platforms besides Linux, so just return `False` if it is on Windows:

`if "webkit" in self.driver_class.lower() and self.exe_platform == "Windows":`